DigitalLearningSolutions.Web/Controllers/TrackingSystem/Centre/Reports/ReportsController.cs

```diff
@@ -19,48 +21,92 @@
+            this.courseDataService = courseDataService;
+            this.courseCategoriesDataService = courseCategoriesDataService;
         }
 
         public IActionResult Index()
         {
             var centreId = User.GetCentreId();
+            var adminId = User.GetAdminId()!.Value;
+            var adminUser = userDataService.GetAdminUserById(adminId)!;
+
+            int? courseCategoryId = null;
+            var validatedCategoryId = adminUser.CategoryId == 0 ? courseCategoryId : adminUser.CategoryId;
 
             var filterData = new ActivityFilterData(
-                DateTime.Today.AddYears(-1),
+                DateTime.UtcNow.Date.AddYears(-1),
                 DateTime.UtcNow,
                 null,
-                null,
+                validatedCategoryId,
                 null,
                 ReportInterval.Months
             );
 
+            Response.Cookies.SetReportsFilterCookie(filterData, DateTime.UtcNow);
+
             var activity = activityService.GetFilteredActivity(centreId, filterData);
 
-            var model = new ReportsViewModel(activity);
+            var jobGroupName = filterData.JobGroupId.HasValue
+                ? jobGroupsDataService.GetJobGroupName(filterData.JobGroupId.Value)
+                : "All";
+            jobGroupName ??= "All";
+
+            var categoryName = filterData.CourseCategoryId.HasValue
+                ? courseCategoriesDataService.GetCourseCategoryName(filterData.CourseCategoryId.Value)
+                : "All";
+            categoryName ??= "All";
+
+            var courseNames = filterData.CustomisationId.HasValue
+                ? courseDataService.GetCourseNameAndApplication(filterData.CustomisationId.Value)
+                : null;
+            var courseNameString = courseNames?.CompositeName ?? "All";
+
+            var filterModel = new ReportsFilterModel(
+                filterData,
+                jobGroupName,
+                categoryName,
+                courseNameString,
+                adminUser.CategoryId == 0
+            );
+
+            var model = new ReportsViewModel(activity, filterModel);
             return View(model);
         }
 
-        [Route("/TrackingSystem/Centre/Reports/Data")]
-        public IEnumerable<ActivityDataRowModel> GetRecentData()
+        [Route("Data")]
+        public IEnumerable<ActivityDataRowModel> GetGraphData()
         {
             var centreId = User.GetCentreId();
-            var filterData = new ActivityFilterData(
-                DateTime.UtcNow.Date.AddYears(-1),
-                DateTime.UtcNow,
-                null,
-                null,
-                null,
-                ReportInterval.Months
-            );
+            var filterData = Request.Cookies.ParseReportsFilterCookie();
 
-            var activity = activityService.GetFilteredActivity(centreId, filterData);
+            var activity = activityService.GetFilteredActivity(centreId, filterData!);

```

Comment:
I feel like there's a possibility that the filter data is not present in the cookie so we should handle the case where it doesn't exist -- we can think about it more when it's all tied together :)

Reply:
Agreed, I was intending to deal with this issue in the middleware. If nothing else we'll need an initialisation system for when a user visits the page for the first time.